### DISCUSSION TITLE
Add OIDC support for self host

### DISCUSSION THREAD

**Discussion by lperin-solvd** [ente-io/ente#2241] [Enhancements]
It would be pretty great to have Open ID Connect support for self hosted servers (like Authentik or Authelia)

**Comment by Arno500** [comment]
That said, it could also be a good opportunity to add social login for the managed version

**Comment by zmberber** [comment]
indeed, SSO is one of the features i am really missing here.

**Reply by JerboaGobi** [reply]
Seconded, SSO support via either OIDC or SAML.

**Comment by Webbanditten** [comment]
Expressing my interest here as well

**Comment by gi-yt** [comment]
+1, it would be a great addition to the project

**Reply by Webbanditten** [reply]
It just makes so much sense. I have loads of family members that if they were to remember another set of passwords would implode.

**Comment by txwgnd** [comment]
+1 for OIDC. It's a feature holding me back to use Ente..

**Comment by Gurkengewuerz** [comment]
+1, would be very cool. ente is almost the last service i am hosting without oidc.

**Comment by nanoandrew4** [comment]
+1 for OICD, I'm migrating several of my self hosted services to use Authentik and just found ente is the only one that does not support OICD or some other form of single login

**Comment by NachoPicchu** [comment]
I believe implementing OIDC in Ente wouldn't be very straightforward. Whoever is in control of the auth server will be able to to authenticate as any user they please (think of the authentik impersonation feature).
Users would need to create a second password within Ente in order to encrypt/decrypt their files.

**Reply by FlexibleToast** [reply]
> I believe implementing OIDC in Ente wouldn't be very straightforward. Whoever is in control of the auth server will be able to to authenticate as any user they please (think of the authentik impersonation feature). Users would need to create a second password within Ente in order to encrypt/decrypt their files.

Yes, that would make sense. However, something similar to how Bitwarden handles this could be done. After a user has entered their encryption password on a device the key could be stored on that local device and locked by the oidc token.

**Comment by helmut72** [comment]
Also came here because I've searched for OIDC and it would be great to have this feature.

**Comment by audricd** [comment]
+1
Immich has SSO..
If ente had it, id migrate to use it with PocketID
